spec/integration/parser/epp_parameters_spec.rb

```diff
@@ -0,0 +1,231 @@
+require 'spec_helper'
+require 'puppet_spec/language'
+
+module PuppetSpec
+module EppRunner
+  extend RSpec::Matchers::DSL
+
+  def self.collect_log(source, args, code)
+    Puppet[:code] = code
+    compiler = Puppet::Parser::Compiler.new(Puppet::Node.new('specification'))
+    logs = []
+    Puppet::Util::Log.with_destination(Puppet::Test::LogCollector.new(logs)) do
+      compiler.compile
+      Puppet::Pops::Evaluator::EppEvaluator.inline_epp(compiler.topscope, source, args)
+    end
+    logs = logs.select { |log| log.level == :notice }.map { |log| log.message }
+    logs
+  end
+
+  def run_expectations(decl, expectations, code = 'undef')
+    expectations.each do |args, result|
+      if result[0].is_a?(Class)
+        it "'#{decl}' called with #{args} should fail with error #{result}" do
+          expect {EppRunner.collect_log(decl, args, code)}.to raise_error(result[0], result[1])
+        end
+      else
+        it "'#{decl}' called with #{args} should produce #{result}" do
+          expect(EppRunner.collect_log(decl, args, code)).to include(*result)
+        end
+      end
+    end
+  end
+end
+
+describe "EPP parameter default expressions" do
+  extend EppRunner
+
+  context "to the left of a parameter can use that parameter's value" do
+    run_expectations(<<-SOURCE.gsub(/^ {6}/, ''),
+      <%|
+        $a = 10,
+        $b = $a
+      |%>
+      <%
+        notice($a)
+        notice($b)
+      %>
+      SOURCE
+      [ [{}, ['10', '10']],
```

Comment:
Here's a sample output from forced failure:
```
  1) EPP parameter default expressions to the left of a parameter can use that parameter's value '<%|
  $a,
  $b = $a
|%>
<%
  notice($a)
  notice($b)
%>
' called with {"a"=>10} should produce ["10", "10"]
```
In what way would the string "when one arg is given, param lookup finds given value" bring more clarity? I understand that the first version was hard to read, unformatted as it was. But now?